FreeBSD: clang: warning: argument unused during compilation: '-pthread'

Might be related to #107 but I'm not sure.
```
[cryptix@somebox ~/go/src/github.com/constabulary/gb:master] uname -a        
FreeBSD somebox.biz 10.1-RELEASE-p9 FreeBSD 10.1-RELEASE-p9 #0: Tue Apr  7 01:09:46 UTC 2015     root@amd64-builder.daemonology.net:/usr/obj/usr/src/sys/GENERIC  amd64
[cryptix@somebox ~/go/src/github.com/constabulary/gb:master] go version      
go version go1.4.2 freebsd/amd64
[cryptix@somebox ~/go/src/github.com/constabulary/gb:master] git log -1      
commit 2fbf2c5653106700910cf5b97adb8fb43fcb6967
Author: Dave Cheney <dave@cheney.net>
Date:   Wed May 20 20:48:04 2015 +1000
  
    move run and runOut to *Context to ensure they are scoped by permit
[cryptix@somebox ~/go/src/github.com/constabulary/gb:master] go test -v ./...
=== RUN TestBuild
a 
a 
b 
a 
d.v1
c 
d.v1
clang: warning: argument unused during compilation: '-pthread'
/usr/bin/ld: unrecognized option '--build-id=none'
/usr/bin/ld: use the --help option for usage information
clang: error: linker command failed with exit code 1 (use -v to see invocation)
clang: warning: argument unused during compilation: '-pthread'
/usr/bin/ld: unrecognized option '--build-id=none'
/usr/bin/ld: use the --help option for usage information
clang: error: linker command failed with exit code 1 (use -v to see invocation)
--- FAIL: TestBuild (4.38s)
        build_test.go:48: ctx.Build(cgomain): want <nil>, got Gc: no Gofiles provided
        build_test.go:48: ctx.Build(cgotest): want <nil>, got Gc: no Gofiles provided
=== RUN TestOneElementCycleDetection
--- PASS: TestOneElementCycleDetection (0.00s)
=== RUN TestSimpleCycleDetection
--- PASS: TestSimpleCycleDetection (0.00s)
=== RUN TestLongCycleDetection
--- PASS: TestLongCycleDetection (0.00s)
=== RUN TestStripext
--- PASS: TestStripext (0.00s)
=== RUN TestResolvePackage
--- PASS: TestResolvePackage (0.00s)
=== RUN TestPackageName
--- PASS: TestPackageName (0.00s)
FAIL
FAIL    github.com/constabulary/gb      4.386s
```

--- a/cgo.go
+++ b/cgo.go
@@ -33,9 +33,12 @@ func rungcc1(ctx *Context, dir, ofile, cfile string) Target {
 // rungcc2 links the o files from rungcc1 into a single _cgo_.o.
 func rungcc2(pkg *Package, dir string, ofile string, ofiles []string) error {
 	args := []string{
-		"-fPIC", "-m64", "-pthread", "-fmessage-length=0",
-		"-o", ofile,
+		"-fPIC", "-m64", "-fmessage-length=0",
 	}
+	if !isClang() {
+		args = append(args, "-pthread")
+	}
+	args = append(args, "-o", ofile)
 	args = append(args, ofiles...)
 	_, _, _, cgoLDFLAGS := cflags(pkg, true)
 	args = append(args, cgoLDFLAGS...) // this has to go at the end, because reasons!
@@ -44,25 +47,29 @@ func rungcc2(pkg *Package, dir string, ofile string, ofiles []string) error {
 
 // rungcc3 links all previous ofiles together with libgcc into a single _all.o.
 func rungcc3(ctx *Context, dir string, ofiles []string) (string, error) {
-	libgcc, err := libgcc(ctx)
-	if err != nil {
-		return "", nil
-	}
 	ofile := filepath.Join(filepath.Dir(ofiles[0]), "_all.o")
 	args := []string{
-		"-fPIC", "-m64", "-pthread", "-fmessage-length=0",
-		"-g", "-O2",
-		"-o", ofile,
+		"-fPIC", "-m64", "-fmessage-length=0",
 	}
+	if !isClang() {
+		args = append(args, "-pthread")
+	}
+	args = append(args, "-g", "-O2", "-o", ofile)
 	args = append(args, ofiles...)
-	args = append(args, "-Wl,-r", "-nostdlib", libgcc, "-Wl,--build-id=none")
+	args = append(args, "-Wl,-r", "-nostdlib")
+	if !isClang() {
+		libgcc, err := libgcc(ctx)
+		if err != nil {
+			return "", nil
+		}
+		args = append(args, libgcc, "-Wl,--build-id=none")
+	}
 	return ofile, ctx.run(dir, nil, gcc(), args...)
 }
 
 // libgcc returns the value of gcc -print-libgcc-file-name.
 func libgcc(ctx *Context) (string, error) {
 	args := []string{
-		"-fPIC", "-m64", "-pthread", "-fmessage-length=0",
 		"-print-libgcc-file-name",
 	}
 	var buf bytes.Buffer
@@ -76,6 +83,10 @@ func cgotool(ctx *Context) string {
 
 func gcc() string {
 	return gccBaseCmd()[0] // TODO(dfc) handle gcc wrappers properly
+}
+
+func isClang() bool {
+	return strings.HasPrefix(gcc(), "clang")
 }
 
 // gccBaseCmd returns the start of the compiler command line.

